crates/m3/tests/merkle_tree.rs

```diff
@@ -557,35 +562,46 @@ mod arithmetization {
 		/// (deduped for multiple paths in the same tree).
 		/// Follows format [Root ID, Digest].
 		pub roots_channel: ChannelId,
+
+		/// Channel for verifying that child depth is one more than parent depth
+		/// has one value.
+		pub lookup_channel: ChannelId,
 	}
 
 	impl MerkleTreeCS {
 		pub fn new(cs: &mut ConstraintSystem) -> Self {
 			let nodes_channel = cs.add_channel("merkle_tree_nodes");
 			let roots_channel = cs.add_channel("merkle_tree_roots");
+			let lookup_channel = cs.add_channel("incr_lookup");
+			let permutation_channel = cs.add_channel("permutation");
 
 			let merkle_path_table_left =
-				NodesTable::new(cs, MerklePathPullChild::Left, nodes_channel);
+				NodesTable::new(cs, MerklePathPullChild::Left, nodes_channel, lookup_channel);
 			let merkle_path_table_right =
-				NodesTable::new(cs, MerklePathPullChild::Right, nodes_channel);
+				NodesTable::new(cs, MerklePathPullChild::Right, nodes_channel, lookup_channel);
 			let merkle_path_table_both =
-				NodesTable::new(cs, MerklePathPullChild::Both, nodes_channel);
+				NodesTable::new(cs, MerklePathPullChild::Both, nodes_channel, lookup_channel);
 
 			let root_table = RootTable::new(cs, nodes_channel, roots_channel);
 
+			let mut table = cs.add_table("incr_lookup_table");
+			let incr_table = IncrLookup::new(&mut table, lookup_channel, permutation_channel, 8);
 			Self {
 				merkle_path_table_left,
 				merkle_path_table_right,
 				merkle_path_table_both,
 				root_table,
 				nodes_channel,
 				roots_channel,
+				lookup_channel,
+				incr_table,
 			}
 		}
 
 		pub fn fill_tables(
 			&self,
 			trace: &MerkleTreeTrace,
+			cs: &ConstraintSystem,
 			witness: &mut WitnessIndex<PackedType<OptimalUnderlier128b, B128>>,
```

Comment:
follow-up: make this generic over U: UnderlierType, not hard-coded to Underlier128b